lib/_http_client.js

```diff
@@ -375,9 +375,6 @@ function _destroy(req, socket, err) {
     socket.destroy(err);
   } else {
     socket.emit('free');
-    if (!req.aborted && !err) {
-      err = connResetException('socket hang up');
```

Comment:
I don't like to remove this.

Reply:
Any specific reason? If I call e.g. `.destroy()` on a `fs` stream it won't cause any error (or any other stream I can think of at the moment).

Is it backwards compat which is you concern?

Reply:
It's an informative error that is being removed for no good reason. What's the problem with it? It doesn't work 100% like `stream.destroy()` but it is really so problematic to justify a semver-major change?

Reply:
I would argue that destroying/aborting a stream is not an error and emitting an error is weird/confusing since for me that would indicate something went wrong.

I do agree that the worth of the change is questionable though.